external-crates/move/crates/move-analyzer/src/completions/name_chain.rs

```diff
@@ -554,56 +554,26 @@ fn struct_completion(
 /// If source module is specified, and it is different from the module of the member,
 /// then only public members are included in the completion, unless modules are
 /// in the same package, in which case package members are also included. No members are included
-/// if a circular dependency is detected.
+/// if a circular dependency is detected. This exclusion is needed to generate
+/// correct copletions for auto-imports.
 fn single_name_member_completion(
     symbols: &Symbols,
     cursor: &CursorContext,
     mod_defs: &ModuleDefs,
     member_alias: &Symbol,
     member_name: &Symbol,
     chain_kind: ChainCompletionKind,
-    src_mod_ident_opt: Option<ModuleIdent>,
 ) -> Vec<CompletionItem> {
     use ChainCompletionKind as CT;
 
-    fn exclude_member(
```

Comment:
Refactored to `utils` as we now use it for both auto-completion and auto-fixes. The use of this function for auto-completions changed changed to be applied earlier (and consistently with how this function is used for auto-fixes), hence more changes in the `exclude_member` function below.